discovery/device.go

```diff
@@ -0,0 +1,77 @@
+package discovery
+
+import (
+	"fmt"
+	"net"
+	"strings"
+)
+
+// NewDevice returns an new chromecast device
+func NewDevice(ip net.IP, port int, properties []string) *Device {
+	return &Device{
+		IP:         ip,
+		Port:       port,
+		Properties: parseProperties(properties),
+	}
+}
+
+// Device represents a device discoverd on the network
+type Device struct {
+	IP         net.IP
+	Port       int
+	Properties map[string]string
+}
+
+// Compatibility with cmd.CastDNSEntry
+func (d Device) GetName() string {
+	return d.Name()
+}
+func (d Device) GetUUID() string {
+	return d.ID()
+}
+func (d Device) GetAddr() string {
+	return d.IP.String()
+}
+func (d Device) GetPort() int {
+	return d.Port
+}
+
+// Kind of more idiomatic methods:
```

Comment:
Feel free to change the `CastDNSEntry` to these. I agree, they are more idiomatic and look nicer. But no need if you'd rather not.

Reply:
The easiest way would be to remove the caching mechanism (which is probably not needed anymore after this PR), but I think this is out of the scope of this PR.

I would rather let it as-is.

Reply:
We won't be able to remove caching since it is still needed when there is more than one cast (which is the likely the more common scenario). 

Fair enough, I will change this later then. Can you remove the comment then as it isn't adding anything?

Reply:
> it is still needed when there is more than one cast

In the case of multiple cast, you have to specify the name (or uuid) of the chromecast you want to target. And if you add the `--first` flag, it can still perform the lookup very fast with the cache disabled (less than 0.5s in my case).

> Can you remove the comment then as it isn't adding anything?

Sure, will do later today.